src/Containers/Impl/Docker/LocalDockerService.cs

```diff
@@ -4,43 +4,57 @@
 using Microsoft.Common.Core.OS;
 using Microsoft.Common.Core.Services;
+using Microsoft.R.Common.Core.Output;
 using Newtonsoft.Json.Linq;
 using static System.FormattableString;
 
 namespace Microsoft.R.Containers.Docker {
     public abstract class LocalDockerService : IDockerService {
+        public const string ContainerOutputName = "R Containers";
         private readonly IProcessServices _ps;
-        private readonly IActionLogWriter _outputLogWriter;
+        private readonly IOutputService _outputService;
         private readonly Regex _containerIdMatcher64 = new Regex("[0-9a-f]{64}", RegexOptions.IgnoreCase);
         private readonly Regex _containerIdMatcher12 = new Regex("[0-9a-f]{12}", RegexOptions.IgnoreCase);
         private readonly int _defaultTimeout = 500;
+        private IOutput _output;
 
         protected LocalDockerService(IServiceContainer services) {
             _ps = services.Process();
-            _outputLogWriter = services.GetService<IActionLogWriter>();
-            // TODO: No instance of IActionLogWriter is exported in default IServiceContainer. Need scope support.
+            _outputService = services.GetService<IOutputService>();
+        }
+        
+        public async Task<IContainer> CreateContainerFromFileAsync(BuildImageParameters buildParams, CancellationToken ct) {
+            await TaskUtilities.SwitchToBackgroundThread();
+
+            var buildOptions = $"-t {buildParams.Image}:{buildParams.Tag} {Path.GetDirectoryName(buildParams.DockerfilePath)}";
+            await BuildImageAsync(buildOptions, ct);
+
+            var createOptions = Invariant($"-p 5444:5444 --name {buildParams.Name} {buildParams.Image}:{buildParams.Tag} rtvsd");
+            var containerId = await CreateContainerAsync(createOptions, ct);
+            return await GetContainerAsync(containerId, ct);
         }
 
         public Task<string> BuildImageAsync(string buildOptions, CancellationToken ct) {
             var command = "build";
-            return ExecuteCommandAsync(Invariant($"{command} {buildOptions}"), -1, true, ct);
+            return ExecuteCommandAsync(Invariant($"{command} {buildOptions}"), command, -1, true, ct);
         }
 
         public async Task<IEnumerable<IContainer>> ListContainersAsync(bool getAll = true, CancellationToken ct = default(CancellationToken)) {
             await TaskUtilities.SwitchToBackgroundThread();
 
             var command = "ps";
             var commandOptions = getAll ? "-a -q" : "-q";
-            var output = await ExecuteCommandAsync(Invariant($"{command} {commandOptions}"), _defaultTimeout, true, ct);
-            var lines = output.Split(new string[] { "\n", "\r\n" }, StringSplitOptions.RemoveEmptyEntries);
+            var output = await ExecuteCommandAsync(Invariant($"{command} {commandOptions}"), null, _defaultTimeout, true, ct);
+            var lines = output.Split(new[] { "\n", "\r\n" }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
I think it will split on `\n` even if it is `\r\n`, no ? Prob doesn't matter here though.

Reply:
@karthiknadig , what do you think?

Reply:
The string actually splits on `\r\n`. The split algorithm compares the first character of the split list against the current character of the string. Then it compares the remaining to determine where it needs to cut. So, if the split list has both `\n` and `\r\n`, it will cut the following string, `sample = "abcd\r\nefgh"`, using `\r\n`, instead of `\n`. Since the current character match when it is at  `sample[4] == '\r'`.
http://referencesource.microsoft.com/#mscorlib/system/string.cs,95f22bdebc0c3230